source/fe/fe_data.cc

```diff
@@ -19,9 +19,193 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+namespace
+{
+  inline unsigned int
+  vertices_per_cell(const ReferenceCell::Type cell_type)
+  {
+    switch (cell_type)
+      {
+        case ReferenceCell::Type::Vertex:
+          return GeometryInfo<0>::vertices_per_cell;
+        case ReferenceCell::Type::Line:
+          return GeometryInfo<1>::vertices_per_cell;
+        case ReferenceCell::Type::Tri:
+          return 3;
+        case ReferenceCell::Type::Quad:
+          return GeometryInfo<2>::vertices_per_cell;
+        case ReferenceCell::Type::Tet:
+          return 4;
+        case ReferenceCell::Type::Hex:
+          return GeometryInfo<3>::vertices_per_cell;
+        default:
+          Assert(false, ExcNotImplemented());
+          return 0;
+      }
```

Comment:
Would it be possible to use `internal::ReferenceCell::Base()` here instead of writing these out?